doc/model.rst

```diff
@@ -0,0 +1,932 @@
+-  `device` ("GPU"): Specifying the device for ML model execution.
+-  `devices_per_node` (2): Use two GPUs per node.
+-  `first_device` (0): Start with 0 index GPU.
+-  `inputs` (inputs): The name of the ML model input nodes (TensorFlow only).
+-  `outputs` (outputs): The name of the ML model output nodes (TensorFlow only).
+
+.. warning::
+    Calling `exp.start(smartsim_model)` prior to instantiation of an orchestrator will result in
+    a failed attempt to load the ML model to a non-existent database.
+
+When the ``Model`` is started via ``Experiment.start()``, the ML model will be loaded to the
+launched ``Orchestrator``. The ML model can then be executed on the ``Orhcestrator`` via a SmartSim
+client (:ref:`SmartRedis<dead_link>`) within the application code.
+
+.. _TS_doc:
+TorchScripts
+============
+When configuring a ``Model``, users can instruct SmartSim to load TorchScripts dynamically
+to the ``Orchestrator``. TorchScripts added are loaded into the ``Orchestrator`` prior to
+the execution of the ``Model``. To load a TorchScript to the database, SmartSim users
+can follow one of the processes:
+
+- :ref:`Define a TorchScript function in-memory<in_mem_TF_doc>`
+   Use the ``Model.add_function()`` to instruct SmartSim to load an in-memory TorchScript to the ``Orchestrator``.
+- :ref:`Define a TorchScript function from file<TS_from_file>`
+   Provide file path to ``Model.add_script()`` to instruct SmartSim to load the TorchScript from file to the ``Orchestrator``.
+- :ref:`Define a TorchScript function as string<TS_raw_string>`
+   Provide function string to ``Model.add_script()`` to instruct SmartSim to load a raw string as a TorchScript function to the ``Orchestrator``.
+
+Continue or select the respective process link to learn more on how each function (``Model.add_script()`` and ``Model.add_function()``)
+dynamically loads TorchScripts to the ``Orchestrator``.
+
+.. _in_mem_TF_doc:
+-------------------------------
+Attach an in-memory TorchScript
+-------------------------------
+Users can define TorchScript functions within the Python driver script
+to attach to a ``Model``. This feature is supported by ``Model.add_function()`` which provides flexible
+device selection, allowing users to choose between which device the the TorchScript is executed on, `"GPU"` or `"CPU"`.
+In environments with multiple devices, specific device numbers can be specified using the
+`devices_per_node` parameter.
+
+.. warning::
+    ``Model.add_function()`` does **not** support loading in-memory TorchScript functions to a colocated ``Orchestrator``.
+    If you would like to load a TorchScript function to a colocated ``Orchestrator``, define the function
+    as a :ref:`raw string<TS_raw_string>` or :ref:`load from file<TS_from_file>`.
+
+When specifying an in-memory TF function using ``Model.add_function()``, the
+following arguments are offered:
+
+- `name` (str): key to store function under.
```

Comment:
key to store function under.

->

reference name for the script inside of the ``Orchestrator``